OpenRA.Mods.Common/Activities/Move/AttackMoveActivity.cs

```diff
@@ -18,31 +19,70 @@ namespace OpenRA.Mods.Common.Activities
 {
 	public class AttackMoveActivity : Activity
 	{
-		const int ScanInterval = 7;
-
+		readonly Func<Activity> getInner;
 		Activity inner;
-		int scanTicks;
+		Activity attack;
 		AutoTarget autoTarget;
+		bool moving;
 
-		public AttackMoveActivity(Actor self, Activity inner)
+		public AttackMoveActivity(Actor self, Func<Activity> getInner)
 		{
-			this.inner = inner;
+			this.getInner = getInner;
 			autoTarget = self.TraitOrDefault<AutoTarget>();
+			moving = false;
 		}
 
 		public override Activity Tick(Actor self)
 		{
-			if (autoTarget != null && --scanTicks <= 0)
+			if (IsCanceled)
 			{
-				autoTarget.ScanAndAttack(self, true);
-				scanTicks = ScanInterval;
+				if (inner != null)
+				{
+					inner = ActivityUtils.RunActivity(self, inner);
+					return this;
+				}
+
+				return NextActivity;
```

Comment:
Shouldn't we also be checking `attack` first? Some activities, like `LeapAttack` are not interruptable.

Reply:
Yes. Added check for attack.